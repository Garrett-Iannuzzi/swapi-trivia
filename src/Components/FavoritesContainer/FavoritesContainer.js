import React from 'react';
import { Link } from 'react-router-dom';
import './FavoritesContainer.scss';
import CharacterCard from '../CharacterCard/CharacterCard.js';
import PropTypes from 'prop-types';


const FavoritesContainer = ({ favorites, handleFavorites, resetCharacters }) => {
  const favoriteCards = favorites.map(character => {
    return(
      <CharacterCard
        {...character}
        key={character.character}
        handleFavorites={handleFavorites}
        resetCharacters={resetCharacters}
      />
    )
  });
  return (
    favorites.length > 0 ?
    <main className='character-container'>
      <Link to='/movies' className='link-back-movies'>
        <button className='button-movies button1' onClick={() => resetCharacters()}>Return to Movies</button>
      </Link>
      <section className="section-character-container">
        {favoriteCards}
      </section>
    </main>
    :
    <main className='no-favorites-main'>
      <Link to='/movies' className='link-back-movies'>
        <button className='button-movies button2' onClick={() => resetCharacters()}>Return to Movies</button>
      </Link>
      <section className='no-favorites-section'>
        <p className='no-favorites-header'>Choose Some Favorite Characters!!!</p>
      </section>
    </main>
  );
}

export default FavoritesContainer;

FavoritesContainer.propTypes = {
  favorites: PropTypes.arrayOf(PropTypes.object),
  handleFavorites: PropTypes.func,
  resetCharacters: PropTypes.func
}
