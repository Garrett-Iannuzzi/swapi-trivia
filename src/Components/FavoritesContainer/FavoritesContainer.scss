.no-favorites-section {
  width: 35%;
  border: indianred 5px solid;
  border-radius: 10px;
  background-color: black;
  color: white;
  height: 50px;
  display: flex;
  justify-content: center;
  margin-top: 50px;
  font-size: 1.5em;
}

.no-favorites-header {
  margin: 0;
}
