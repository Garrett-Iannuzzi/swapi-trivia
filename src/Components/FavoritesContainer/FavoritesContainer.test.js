import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import FavoritesContainer from '../FavoritesContainer/FavoritesContainer.js';

describe('FavoritesContainer', () => {

  let wrapper;
  const mockHandleFavorites = jest.fn();
  const mockResetCharacter = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<FavoritesContainer
        favorites={[{
            character: "C-3PO",
            creature: "artificial",
            name: "Tatooine",
            openingCrawl: "the guardians of",
            population: "200000",
            relatedFilms:[
              {relatedFilms: "The Empire Strikes Back"},
              {relatedFilms: "Attack of the Clones"}],
              species: "Droid"
          }, {
            character: "R2-D2",
            creature: "artificial",
            name: "Naboo",
            openingCrawl: "Turmoil has engulfed",
            population: "4500000000",
            relatedFilms: [
              {relatedFilms: "The Empire Strikes Back"},
              {relatedFilms: "Attack of the Clones"}],
            species: "Droid"
          }]}
        handleFavorites={mockHandleFavorites}
        resetCharacters={mockResetCharacter}
      />);
  })

  it('Should match the snapshot with all data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call resetCharacters on click', () => {
    wrapper.find('.button1').simulate('click');
    expect(mockResetCharacter).toHaveBeenCalled();
  });

  it('Should call resetCharacters on click when favorite.length === 0', () => {
    wrapper = shallow(<FavoritesContainer
        favorites={[]}
        handleFavorites={mockHandleFavorites}
        resetCharacters={mockResetCharacter}
      />);
    wrapper.find('.button2').simulate('click');
    expect(mockResetCharacter).toHaveBeenCalled();
  });
})
