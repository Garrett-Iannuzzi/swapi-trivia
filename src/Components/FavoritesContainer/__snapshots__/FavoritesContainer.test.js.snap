// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesContainer Should match the snapshot with all data 1`] = `
<main
  className="character-container"
>
  <Link
    className="link-back-movies"
    to="/movies"
  >
    <button
      className="button-movies button1"
      onClick={[Function]}
    >
      Return to Movies
    </button>
  </Link>
  <section
    className="section-character-container"
  >
    <CharacterCard
      character="C-3PO"
      creature="artificial"
      handleFavorites={[MockFunction]}
      key="C-3PO"
      name="Tatooine"
      openingCrawl="the guardians of"
      population="200000"
      relatedFilms={
        Array [
          Object {
            "relatedFilms": "The Empire Strikes Back",
          },
          Object {
            "relatedFilms": "Attack of the Clones",
          },
        ]
      }
      species="Droid"
    />
    <CharacterCard
      character="R2-D2"
      creature="artificial"
      handleFavorites={[MockFunction]}
      key="R2-D2"
      name="Naboo"
      openingCrawl="Turmoil has engulfed"
      population="4500000000"
      relatedFilms={
        Array [
          Object {
            "relatedFilms": "The Empire Strikes Back",
          },
          Object {
            "relatedFilms": "Attack of the Clones",
          },
        ]
      }
      species="Droid"
    />
  </section>
</main>
`;
